client/lecturer/app/engagement.directive.js

```diff
@@ -19,9 +19,32 @@
         function link(scope, element, attrs, engagementCanvas) {
             console.log('Ready (Engagement Canvas link)');
 
-            engagementCanvas.canvas = element[0];
+            engagementCanvas.width = element.width();
+            engagementCanvas.height = element.height();
+
+            engagementCanvas.canvas = document.createElement('canvas');
+            element.append(engagementCanvas.canvas);
             engagementCanvas.ctx = engagementCanvas.canvas.getContext('2d');
 
+            engagementCanvas.canvas.width = engagementCanvas.width;
+            engagementCanvas.canvas.height = engagementCanvas.height;
```

Comment:
Hm. It might be fine without the engagementCanvas.width since it is only used for the background canvas, I think. I'll look at it and change it.